python/cudf/cudf/_lib/table.pyx

```diff
@@ -48,6 +48,10 @@ cdef class Table:
     def _num_columns(self):
         return len(self._data)
 
+    @property
+    def _num_indices(self):
+        return len(self._index_names)
```

Comment:
This can end up throwing a confusing error if `_index_names` is `None`. Suggest checking for None and throwing a `ValueError`

Reply:
Ah right. Good catch

Reply:
Should we actually return `0` in that case?

Reply:
Yes, good call!